### DISCUSSION TITLE
Upsert

### DISCUSSION THREAD

**Discussion by commongeek** [rqlite/rqlite#1875] [Q&A] [answered]
Does rqlite support upserts (https://www.sqlite.org/lang_upsert.html)?

**Comment by otoolep** [comment]
Sure, no reason it wouldn't. But here's unit test anyway, just to be sure:

https://github.com/rqlite/rqlite/pull/1876

**Reply by commongeek** [reply]
Thanks for your answer. Does this mean that I can send all sqlite commands via HTTP API, e.g. create view, create trigger etc.?

**Reply by otoolep** [reply]
Yes.

**Comment by otoolep** [comment, marked as answer]
Generally, yes, with some important exceptions. You should not manage transactions explicitly (`BEGIN`, `COMMIT`, etc) and some non-deterministic functions may not behave the way you expect. See https://rqlite.io/docs/api/non-deterministic/

Apart from that everything else should work fine.

**Comment by otoolep** [comment]
If you see behaviour you don't understand, open an issue with the details and I'll look into it.
